### DISCUSSION TITLE
RLS + nextjs getStaticProps setup on a dynamic page

### DISCUSSION THREAD

**Discussion by zidkim** [supabase/supabase#6665] [Questions] [answered]
Hello All,

I'm trying to fetch data inside of getStaticProps on a dynamically generated page. To be clear, by [dynamic](https://nextjs.org/docs/basic-features/data-fetching/get-static-paths) I mean dynamic routes using getStaticPaths combined with getStaticProps.
I have RLS enabled to accept authenticated users.

I understand that given the HTML is generated at build time in getStaticProps, there is no access to req/res.
Is there a good workaround? Or is getServerSideProps the only way?

**Comment by liaujianjie** [comment, marked as answer]
This question has to do more with Next than with Supabase.

`getStaticProps` is not meant to be used for user-specific data. It says in the [Next docs](https://nextjs.org/docs/basic-features/data-fetching/get-static-paths#when-should-i-use-getstaticpaths):

> ### [When should I use getStaticPaths?](https://nextjs.org/docs/basic-features/data-fetching/get-static-paths#when-should-i-use-getstaticpaths)
> - The data comes from a headless CMS
> - The data comes from a database
> - The data comes from the filesystem
> - **The data can be publicly cached (not user-specific)**
> - The page must be pre-rendered (for SEO) and be very fast — `getStaticProps` generates `HTML` and `JSON` files, both of which can be cached by a CDN for performance

So yes, `getServerSideProps` is the only way.

**Reply by liaujianjie** [reply]
If you mean to have build-time SSG-ed pages but with runtime SSR-ed authenticated pages for the same pages:

Unfortunately, it's currently not possible to use `getStaticProps` and `getServerSideProps` at the same time. [There's a discussion on this over at the Next repo.](https://github.com/vercel/next.js/discussions/11424)

**Reply by zidkim** [reply]
Hi @liaujianjie, thank you for confirming.
Perhaps this should be a feature request, as I'm able to freely use Firebase SDK inside of getStaticProps/getStaticPaths, and this works OOB. I suppose this has more to do with how Firebase handles security using Firebase rules.
